drivers/wifi/nrf700x/zephyr/src/zephyr_ps.c

```diff
@@ -60,3 +60,52 @@ int wifi_nrf_set_power_save(const struct device *dev,
 out:
 	return ret;
 }
+
+
+int wifi_nrf_set_power_save_mode(const struct device *dev,
+				 struct wifi_ps_mode_params *ps_mode_params)
+{
+	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
+	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
+	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = NULL;
+	int ret = -1;
+
+	if (!dev || !ps_mode_params) {
+		goto out;
+	}
+
+	vif_ctx_zep = dev->data;
+
+	if (!vif_ctx_zep) {
+		LOG_ERR("%s: vif_ctx_zep is NULL\n", __func__);
+		goto out;
+	}
+
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+
+	if (!rpu_ctx_zep) {
+		LOG_ERR("%s: rpu_ctx_zep is NULL\n", __func__);
+		goto out;
+	}
+
+	if (ps_mode_params->mode == WIFI_PS_MODE_WMM) {
+		/*WMM mode */
+		status = wifi_nrf_fmac_set_uapsd_queue(rpu_ctx_zep->rpu_ctx,
```

Comment:
Though not in this commit, API looks tied to Linux style, can we use LEGACY/WMM and internally to UMAC use uapsd_queues? If its too much rework then ignore.